### PR TITLE
Hal API to retrieve TID stats via iw and Netlink, Adding Hal test binary

### PR DESCRIPTION
**Pull request by Rajakumaran1106**
Signed-off-by: Rajakumaran1106 <rajakumaran.a@ltts.com>

### CODE REVIEW

**Comment by manigandanj** [comment]
@Rajakumaran1106 , 
Please address review comments as part of PR https://github.com/rdkcentral/rdkb-turris-hal/pull/33

File: source/wifi/Makefile.am [hunk 1]
```diff
@@ -28,3 +28,8 @@ lib_LTLIBRARIES = libhal_wifi.la
 libhal_wifi_la_CPPFLAGS = -I$(top_srcdir)/./include $(CPPFLAGS)
```
Comment (manigandanj) [inline source/wifi/Makefile.am:28]: remove this "./"

File: source/wifi/Makefile.am [hunk 2]
```diff
@@ -28,3 +28,8 @@ lib_LTLIBRARIES = libhal_wifi.la
...
 libhal_wifi_la_SOURCES = wifi_hal.c
 libhal_wifi_la_LDFLAGS =
+
+#wifihal test binary
+bin_PROGRAMS = wifihal
+wifihal_CPPFLAGS = -I$(top_srcdir)/./include $(CPPFLAGS)
```
Comment (manigandanj) [inline source/wifi/Makefile.am:34]: remove this "./"

File: source/wifi/wifi_hal.c [hunk 3]
```diff
@@ -7282,14 +7469,15 @@ INT wifi_chan_eventRegister(void *eventCb)
...
 int main(int argc,char **argv)
 {
-        WIFI_ENTRY_EXIT_DEBUG("Inside %s:%d\n",__func__, __LINE__);
-	int index;
-	INT ret=0;
+    WIFI_ENTRY_EXIT_DEBUG("Inside %s:%d\n",__func__, __LINE__);
+    int index;
+    INT ret=0;
     if(argc <= 1) {
         printf("help\n");
```
Comment (manigandanj) [inline source/wifi/wifi_hal.c:7479]: do we need this line.

File: source/wifi/wifi_hal.c [hunk 4]
```diff
@@ -7301,8 +7489,20 @@ int main(int argc,char **argv)
     else if(strstr(argv[1], "reset")!=NULL) {
         return wifi_reset();
     }    
-	
-	index = atoi(argv[2]);
+    index = atoi(argv[2]); //radio index
+
+    if(strstr(argv[1],"wifi_getApAssociatedDeviceTidStatsResult")!=NULL)
+    {
+        printf(" %s %s %s \n", argv[1],argv[2],argv[3]);
```
Comment (manigandanj) [inline source/wifi/wifi_hal.c:7496]: Group all declaration on top inside this if block.

File: source/wifi/wifi_hal.c [hunk 5]
```diff
@@ -7301,8 +7489,20 @@ int main(int argc,char **argv)
...
+        char sta[20] = {'\0'};
+        u64 handle= 0;
+        strcpy(sta,argv[3]);
+        mac_address_t *st;
+        st=sta;
+
+        wifi_associated_dev_tid_stats_t tid_stats;
+        wifi_getApAssociatedDeviceTidStatsResult(index,st,&tid_stats,handle);
```
Comment (manigandanj) [inline source/wifi/wifi_hal.c:7504]: After calling HAL API, print the results. This is very important.

File: source/wifi/wifi_hal.c [hunk 6]
```diff
@@ -7189,7 +7377,6 @@ INT wifi_getRadioChannelStats(INT radioIndex,wifi_channelStats_t *input_output_c
     nl_cb_set(nl.cb,NL_CB_VALID,NL_CB_CUSTOM,chanSurveyInfo_callback,local);
     nl_recvmsgs(nl.socket, nl.cb);
     nlmsg_free(msg);
-    nl_cb_put(nl.cb);
     nlfree(&nl);
     //Copying the Values
     for(int i=0;i<=array_size;i++)
```
Comment (manigandanj) [inline source/wifi/wifi_hal.c:7382]: Looks like, there is indent mistake in this for loop.

File: source/wifi/wifi_hal.c [hunk 7]
```diff
@@ -7189,7 +7377,6 @@ INT wifi_getRadioChannelStats(INT radioIndex,wifi_channelStats_t *input_output_c
     nl_cb_set(nl.cb,NL_CB_VALID,NL_CB_CUSTOM,chanSurveyInfo_callback,local);
     nl_recvmsgs(nl.socket, nl.cb);
     nlmsg_free(msg);
-    nl_cb_put(nl.cb);
```
Comment (manigandanj) [inline source/wifi/wifi_hal.c:7192]: why this line is remove?

File: source/wifi/wifi_hal.c [hunk 8]
```diff
@@ -3181,7 +3203,6 @@ INT wifi_getAssociatedDeviceDetail(INT apIndex, INT devIndex, wifi_device_t *out
     nl_cb_set(nl.cb,NL_CB_VALID,NL_CB_CUSTOM,AssoDevInfo_callback,&info);
     nl_recvmsgs(nl.socket, nl.cb);
     nlmsg_free(msg);
-    nl_cb_put(nl.cb);
```
Comment (manigandanj) [inline source/wifi/wifi_hal.c:3184]: why this line is remove?

**Review by manigandanj — CHANGES_REQUESTED** [review]

File: source/wifi/Makefile.am [hunk 9]
```diff
@@ -28,3 +28,8 @@ lib_LTLIBRARIES = libhal_wifi.la
 libhal_wifi_la_CPPFLAGS = -I$(top_srcdir)/./include $(CPPFLAGS)
 libhal_wifi_la_SOURCES = wifi_hal.c
 libhal_wifi_la_LDFLAGS =
+
+#wifihal test binary
+bin_PROGRAMS = wifihal
```
Comment (manigandanj) [inline source/wifi/Makefile.am:33]: This block is not inside any conditional.

File: source/wifi/wifi_hal.c (hunk 8 above)
Comment (Rajakumaran1106) [reply source/wifi/wifi_hal.c:3184]: nl_cb_put  has been  handled in nlfree() function, so removing the extra line

**Review by manigandanj — CHANGES_REQUESTED** [review]

**Pull request closed without merging**